dashboard/lib/services/curriculum_pdfs.rb

```diff
@@ -1,22 +1,52 @@
 require 'cdo/chat_client'
 require 'dynamic_config/dcdo'
 
-# Overview of all Curriculum PDFs
+# This module is the entry point for all "Curriculum PDFs", referring to the
+# PDFs that we automatically generate from our teacher-facing Lesson Plans and
+# related content.
 #
-# Current:
-# - teacher lesson plans (csp1-2021/20210216001309/teacher-lesson-plans/Welcome to CSP.pdf)
-# - student lesson plans (csp1-2021/20210216001309/student-lesson-plans/Welcome to CSP.pdf)
-# - script resources (csp1-2021/20210216001309/resources/CSP Unit 1 - Digital Information ('21-'22) - Resources.pdf)
-# - script overview (csp1-2021/20210216001309/CSP Unit 1 - Digital Information ('21-'22).pdf)
+# Note that, following Rails convention, this file defines the basic structure
+# of the module but most of the logic lives in the submodules, as defined in
+# the various other files in the `curriculum_pdfs/` directory.
+#
+# Currently (as of September 2021), we generate the following kind of PDFs:
+#
+# - Teacher-facing Lesson Plans (for example, https://lesson-plans.code.org/csp1-2021/20210826162223/teacher-lesson-plans/Welcome+to+CSP.pdf)
+# - Student-facing Lesson Plans (for example, https://lesson-plans.code.org/csp1-2021/20210826162223/student-lesson-plans/Welcome+to+CSP.pdf)
+# - Script Overviews (for example, https://lesson-plans.code.org/csp1-2021/20210826162223/Digital+Information+('21-'22).pdf)
+# - Script Resource Rollups (for example, https://lesson-plans.code.org/csp1-2021/20210826162223/Digital+Information+('21-'22)+-+Resources.pdf)
+#
+# We may also want in the future to generate more kinds, including:
 #
-# Future:
 # - student resources, based on Resource#audience
 # - unit calendar
 # - four rollup pages for unit, unit group, one for each of:
 #   - vocab
 #   - resouces
 #   - standards
 #   - programming expressions
+#
+# Our overall approach for PDF generation is as follows:
+#
+# When lesson plan is updated on levelbuilder, we serialize it into a
+# `.script_json` file with an `updated_at` timestamp. When that file is then
+# seeded on the staging machine, we compare that timestamp to the last known
+# value; if we detect that any content in the entire `.script_json` file has
+# been updated, we regenerate all PDFs for the associated script.
+#
+# PDFs are generated by puppeteer. Specifically, all content that is going to
+# be converted into a PDF is built with CSS print styles as well as web-view
```

Comment:
Can you leave a note that this also means that we only generate the page as it looks to a signed out user?